mm2src/coins/eth.rs

```diff
@@ -2420,6 +2435,128 @@ impl MmCoin for EthCoin {
         }))
     }
 
+    fn get_sender_trade_fee(
+        &self,
+        value: TradePreimageValue,
+    ) -> Box<dyn Future<Item = TradeFee, Error = TradePreimageError> + Send> {
+        let coin = self.clone();
+        let fut = async move {
+            let gas_price = try_map!(coin.get_gas_price().compat().await, TradePreimageError::Other);
+            let gas_price = increase_by_percent_one_gwei(gas_price, TRADE_PREIMAGE_GAS_PRICE_PERCENT);
+            let gas_limit = match coin.coin_type {
+                EthCoinType::Eth => {
+                    // this gas_limit includes gas for `ethPayment` and `senderRefund` contract calls
+                    U256::from(300_000)
+                },
+                EthCoinType::Erc20(_) => {
+                    let value = match value {
+                        TradePreimageValue::Exact(value) | TradePreimageValue::UpperBound(value) => {
+                            try_map!(wei_from_big_decimal(&value, coin.decimals), TradePreimageError::Other)
+                        },
+                    };
+                    let allowed = try_map!(
+                        coin.allowance(coin.swap_contract_address).compat().await,
+                        TradePreimageError::Other
+                    );
+                    if allowed < value {
+                        // this gas_limit includes gas for `approve`, `erc20Payment` and `senderRefund` contract calls
+                        U256::from(450_000)
```

Comment:
We can reduce this as an `approve` call typically requires about 50k of gas, if I'm not mistaken.

Reply:
[Here](https://github.com/KomodoPlatform/atomicDEX-API/blob/mm2.1/mm2src/coins/eth.rs#L1447) we specify the 150k of gas for performing the `approve` contract call.